service/worker/scanner/shardscanner/activities.go

```diff
@@ -0,0 +1,336 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package shardscanner
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+
+	"go.uber.org/cadence"
+	"go.uber.org/cadence/.gen/go/shared"
+	"go.uber.org/cadence/activity"
+
+	c "github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/persistence"
+	"github.com/uber/cadence/common/reconciliation/store"
+)
+
+const (
+	// ActivityScannerEmitMetrics is the activity name for ScannerEmitMetricsActivity
+	ActivityScannerEmitMetrics = "cadence-sys-executions-shardscanner-emit-metrics-activity"
+	// ActivityScannerConfig is the activity name ScannerConfigActivity
+	ActivityScannerConfig = "cadence-sys-executions-shardscanner-config-activity"
+	// ActivityScanShard is the activity name for ScanShardActivity
+	ActivityScanShard = "cadence-sys-executions-shardscanner-scan-shard-activity"
+	// ActivityFixerCorruptedKeys is the activity name for FixerCorruptedKeysActivity
+	ActivityFixerCorruptedKeys = "cadence-sys-executions-shardfixer-corrupted-keys-activity"
+	// ActivityFixShard is the activity name for FixShardActivity
+	ActivityFixShard = "cadence-sys-executions-shardfixer-fix-shard-activity"
+	// ShardCorruptKeysQuery is the query name for the query used to get all completed shards with at least one corruption
+	ShardCorruptKeysQuery = "shard_corrupt_keys"
+)
+
+// ScannerConfigActivity will read dynamic config, apply overwrites and return a resolved config.
+func ScannerConfigActivity(
+	activityCtx context.Context,
+	params ScannerConfigActivityParams,
+) (ResolvedScannerWorkflowConfig, error) {
+	ctx := activityCtx.Value(params.ContextKey).(Context)
+	dc := ctx.Config.DynamicParams
+
+	result := ResolvedScannerWorkflowConfig{
+		GenericScannerConfig: GenericScannerConfig{
+			Enabled:                 dc.Enabled(),
+			Concurrency:             dc.Concurrency(),
+			PageSize:                dc.PageSize(),
+			BlobstoreFlushThreshold: dc.BlobstoreFlushThreshold(),
+			ActivityBatchSize:       dc.ActivityBatchSize(),
+		},
+	}
+
+	if ctx.Hooks != nil && ctx.Hooks.Config != nil {
+		result.CustomScannerConfig = ctx.Hooks.Config(ctx)
```

Comment:
Is config the right name for this? These actually contain callback functions right?

Reply:
renamed to `GetScannerConfig`